rest-api-spec/src/main/resources/rest-api-spec/test/indices.data_stream/10_basic.yml

```diff
@@ -54,3 +54,51 @@
 
   - match: { status: 400 }
   - match: { error.root_cause.0.type: "illegal_argument_exception" }
+
+---
+"Delete data stream with backing indices":
+  - skip:
+      version: " - 7.99.99"
+      reason: "enable in 7.8+ after back-porting"
+
+  - do:
+      indices.create_data_stream:
+        name: delete-data-stream1
+        body:
+          timestamp_field: "@timestamp"
+  - is_true: acknowledged
+
+  - do:
+      indices.create:
+        index: test_index
+        body:
+          settings:
+            number_of_shards:   1
+            number_of_replicas: 1
+
+  - do:
+      indices.get:
+        index: "*"
+        expand_wildcards: all
+
+  - is_true: test_index.settings
+  - is_true: delete-data-stream1-000001.settings
+
+  - do:
+      indices.get_data_streams: {}
+  - match: { 0.name: delete-data-stream1 }
+  - match: { 0.timestamp_field: '@timestamp' }
+  - length: { 0.indices: 1 }
+  - match: { 0.indices.0.index_name: 'delete-data-stream1-000001' }
+
+  - do:
+      indices.delete_data_stream:
+        name: delete-data-stream1
+  - is_true: acknowledged
+
+  - do:
+      indices.get:
+        index: "*"
+        expand_wildcards: all
+
+  - is_true: test_index.settings
```

Comment:
I think a validation that the index is gone is missing?

Also, I would find it easier to validate that the index exists using something like:

```
indices.get:
  index: "test_index"
```

and the same for verifying that it is gone, but expect an exception?